crates/next-core/js/src/internal/chunk-loader.ts

```diff
@@ -0,0 +1,84 @@
+// Adapted from https://github.com/vercel/next.js/blob/canary/packages/next/client/route-loader.ts
+
+import { markAssetError } from "next/dist/client/route-loader";
+
+const loadedScripts: Map<string, Promise<unknown>> = new Map();
+const styleSheets: Map<string, Promise<HTMLLinkElement>> = new Map();
+
+export function maybeExecuteScript(src: string): Promise<unknown> {
```

Comment:
Please add a small doc comment to this and `maybeLoadStylesheet` to explain why they're conditional.